pkg/issuer/ca/issue.go

```diff
@@ -128,10 +128,18 @@ func (c *CA) obtainCertificate(crt *v1alpha1.Certificate, signeeKey interface{},
 	}
 
 	// sign and encode the certificate
-	crtPem, _, err := pki.SignCertificate(template, signerCert, signeeKey, signerKey)
+	crtPem, _, err := pki.SignCertificate(template, signerCertChain[0], signeeKey, signerKey)
 	if err != nil {
 		return nil, err
 	}
 
+	// encode the chain
+	chainPem, err := pki.EncodeX509Chain(signerCertChain)
```

Comment:
So it looks like `signerCertChain` includes the full chain, including the 'root' certificate. If for example the CA here is a self signed root, we'd be bundling that with all leaf certificates generated from that root.

From my understanding, this is *not* desired behaviour, although I may be wrong. Are there any other similar examples or info on how this usually behaves elsewhere?

Reply:
Hmm. I don't think we need/want the self signed root. I don't think it makes any difference though - every implementation I've seen, all of the chain certs get imported into a cert pool that can be used to help verify. Importing the root there makes no difference to the outcome.

(In golang for example a verify operation has a certPool of certs that are trusted, and another certPool of arbitrary certs that can be used to form chains to the trusted ones.)

Reply:
@munnerz I've made it so the chain will now exclude self-signed certificates. How's that?